src/core/SampleBuffer.cpp

```diff
@@ -214,54 +199,49 @@ void SampleBuffer::update( bool _keep_settings )
 			// workaround for a bug in libsndfile or our libsndfile decoder
 			// causing some OGG files to be distorted -> try with OGG Vorbis
 			// decoder first if filename extension matches "ogg"
-			if( m_frames == 0 && fileInfo.suffix() == "ogg" )
+			if( dataFrames == 0 && fileInfo.suffix() == "ogg" )
 			{
-				m_frames = decodeSampleOGGVorbis( f, buf, channels, samplerate );
+				dataFrames = decodeSampleOGGVorbis( f, buf, channels, samplerate );
 			}
 #endif
-			if( m_frames == 0 )
+			if( dataFrames == 0 )
 			{
-				m_frames = decodeSampleSF( f, fbuf, channels,
+				dataFrames = decodeSampleSF( f, fbuf, channels,
 									samplerate );
 			}
 #ifdef LMMS_HAVE_OGGVORBIS
-			if( m_frames == 0 )
+			if( dataFrames == 0 )
 			{
-				m_frames = decodeSampleOGGVorbis( f, buf, channels,
+				dataFrames = decodeSampleOGGVorbis( f, buf, channels,
 									samplerate );
 			}
 #endif
-			if( m_frames == 0 )
+			if( dataFrames == 0 )
 			{
-				m_frames = decodeSampleDS( f, buf, channels,
+				dataFrames = decodeSampleDS( f, buf, channels,
 									samplerate );
 			}
 
 			delete[] f;
 		}
 
-		if ( m_frames == 0 || fileLoadError )  // if still no frames, bail
+		if ( dataFrames == 0 || fileLoadError )  // if still no frames, bail
 		{
 			// sample couldn't be decoded, create buffer containing
 			// one sample-frame
-			m_data = MM_ALLOC( sampleFrame, 1 );
-			memset( m_data, 0, sizeof( *m_data ) );
-			m_frames = 1;
+			m_data.clear ();
 			m_loopStartFrame = m_startFrame = 0;
 			m_loopEndFrame = m_endFrame = 1;
```

Comment:
I think you should set to `m_endFrame` and `m_loopEndFrame` to 0 if the buffer is empty.